rrmngmnt/host.py

```diff
@@ -233,11 +242,15 @@ def executor(self, user=None, pkey=False, sudo=False):
             ef = copy.copy(self.executor_factory)
             ef.use_pkey = pkey
             return ef.build(self, user, self.sudo)
-        return self.executor_factory.build(self, user, sudo=self.sudo)
+        if sock:
```

Comment:
I think we don't need `self.sock` in this class.

I would keep it only in Executor, and in ExecutorFactory